deepchecks/vision/vision_data/utils.py

```diff
@@ -0,0 +1,121 @@
+# Copyright (C) 2021-2022 Deepchecks (https://www.deepchecks.com)
+#
+# This file is part of Deepchecks.
+# Deepchecks is distributed under the terms of the GNU Affero General
+# Public License (version 3 or later).
+# You should have received a copy of the GNU Affero General Public License
+# along with Deepchecks.  If not, see <http://www.gnu.org/licenses/>.
+# ----------------------------------------------------------------------------
+#
+"""Utils module for VisionData functionalities."""
+import typing as t
+from collections import Counter
+from enum import Enum
+from numbers import Number
+
+import numpy as np
+import torch
+
+
+class TaskType(Enum):
+    """Enum containing supported task types."""
+
+    CLASSIFICATION = 'classification'
+    OBJECT_DETECTION = 'object_detection'
+    SEMANTIC_SEGMENTATION = 'semantic_segmentation'
+    OTHER = 'other'
+
+
+class BatchOutputFormat(t.TypedDict):
+    images: t.Optional[t.Union[np.ndarray, t.Sequence]]
+    labels: t.Optional[t.Union[np.ndarray, t.Sequence]]
+    predictions: t.Optional[t.Union[np.ndarray, t.Sequence]]
+    additional_data: t.Optional[t.Union[np.ndarray, t.Sequence]]
+    embeddings: t.Optional[t.Union[np.ndarray, t.Sequence]]
+    image_identifiers: t.Optional[t.Union[np.ndarray, t.Sequence]]
+
+
+def sequence_to_numpy(data: t.Optional[t.Sequence], expected_dtype=None) -> t.Optional[t.Sequence]:
+    """Convert a sequence containing some type of array to a List of numpy arrays.
+
+    Returns
+    -------
+    t.Optional[t.Sequence]
+        A list of numpy arrays of the provided data.
+    """
+    if data is None:
+        return None
+    return [object_to_numpy(x, expected_dtype) for x in data]
```

Comment:
This seems like a potentially big degradation in efficiency for batch operations. 
If the dimensions of all x in data match, why not produce a higher dimensional numpy array by concating?

Reply:
I dont think it will have a big effect do to generality of the checks but lets discuss